jax/_src/numpy/ufuncs.py

```diff
@@ -423,9 +455,44 @@ def expm1(x: ArrayLike, /) -> Array:
   """
   return lax.expm1(*promote_args_inexact('expm1', x))
 
-@implements(np.log1p, module='numpy')
+
 @partial(jit, inline=True)
 def log1p(x: ArrayLike, /) -> Array:
+  """Calculates element-wise logarithm of one plus input, ``log(x+1)``.
+
+  JAX implementation of :obj:`numpy.log1p`.
+
+  Args:
+    x: input array or scalar.
+
+  Returns:
+    An array containing the logarithm of one plus of each element in ``x``,
+    promotes to inexact dtype.
+
+  Note:
+    ``jnp.log1p`` has much higher precision than the naive computation of
+    ``log(x+1)`` for small values of ``x``.
+
+  See also:
+    - :func:`jax.numpy.expm1`: Calculates :math:`e^x-1` of each element of the
+      input.
+    - :func:`jax.numpy.log2`: Calculates base-2 logarithm of each element of input.
+    - :func:`jax.numpy.log`: Calculates element-wise logarithm of the input.
+
+  Examples:
+    >>> x = jnp.array([2, 5, 9, 4])
+    >>> jnp.allclose(jnp.log1p(x), jnp.log(x+1))
+    Array(True, dtype=bool)
+
+    For values very close to 0, ``jnp.log1p(x)`` is much more accurate than
+    ``jnp.log(x+1)``:
+
+    >>> x1 = jnp.array([1e-4, 1e-6, 2e-10])
+    >>> jnp.log1p(x1)
+    Array([9.999500e-05, 9.999995e-07, 2.000000e-10], dtype=float32)
+    >>> jnp.log(x1+1)
+    Array([1.00011595e-04, 9.53673862e-07, 0.00000000e+00], dtype=float32)
```

Comment:
How a reader would see that `log1p` result is more accurate than of `log`? Perhaps use
`jnp.expm1(jnp.log1p(x1))` and `jnp.expm1(jnp.log(x1+1))` to demonstrate that log1p/expm1 indeed are more accurate when x1 is close to 0 as these are able to reproduce the input `x1`.

Reply:
Thanks for the suggestion. Updated the example as suggested.